test/builtin_dataset_mocks.py

```diff
@@ -129,10 +133,31 @@ def load(self, config, *, decoder=DEFAULT_DECODER):
         return datapipe, mock_info
 
 
+def config_id(name, config):
```

Comment:
Would the default pytest id generation be enough here?

Reply:
Unfortunately not. You would get something like

```
test/test_prototype_builtin_datasets.py::TestCommon::test_smoke[dataset_mock0-config0]
test/test_prototype_builtin_datasets.py::TestCommon::test_smoke[dataset_mock1-config1]
test/test_prototype_builtin_datasets.py::TestCommon::test_smoke[dataset_mock2-config2]
...
```

The problem is that by default, `pytest` uses the name of the parameter as well as a running index as id. So if we want to have somewhat expressive test names, we need to set the id ourselves. The same example as above but with the custom id from this PR looks like

```
test/test_prototype_builtin_datasets.py::TestCommon::test_smoke[mnist-train]
test/test_prototype_builtin_datasets.py::TestCommon::test_smoke[mnist-test]
test/test_prototype_builtin_datasets.py::TestCommon::test_smoke[fashionmnist-train]
...
```